### ISSUE TITLE
How to change the Font?

### ISSUE THREAD

**Issue by ConAntares** [mbaz/Gaston.jl#112]
I want to use "CMU-Serif" font. But I don't know how to set it with Gaston.

**Comment by mbaz** [comment]
With Gaston v0.9.2, you can do it as follows:

julia> set(print_fonface="CMU-Serif")
julia> plot(rand(10))  # will use the selected font

The current master branch has a lot of fixes and improved documentation; I'll release it as soon as I fix a few loose ends. In master, you can do:

julia> plot(rand(10), font="CMU-Serif")

and you can set the default font globally with

julia> set(font="CMU-Serif")

**Comment by ConAntares** [comment]
> 
> 
> With Gaston v0.9.2, you can do it as follows:
> 
> ```
> julia> set(print_fontface="CMU-Serif")
> julia> plot(rand(10))  # will use the selected font
> ```
> 
> The current master branch has a lot of fixes and improved documentation; I'll release it as soon as I fix a few loose ends. In master, you can do:
> 
> `julia> plot(rand(10), font="CMU-Serif")`
> 
> and you can set the default font globally with
> 
> `julia> set(font="CMU-Serif")`

However, I have attempted ```set(font="CMU-Serif")```  I got the Error: ```no method matching set(; font="CMU-Serif")```

I think Gnuplot is more powerful than Pyplot in the scientific drawing. I am looking forward to your update very much. :)

**Comment by ConAntares** [comment]
My version of Gaston is already v0.9.2, and my version of Julia is 1.1 working in Windows10.
All these ways were failed, :(.
```
using Gaston
t = 0:0.01:1
set(print_fonface="CMU-Serif")
plot(t, sin.(2π*5*t), title = "A sine wave")
```
```
got unsupported keyword argument "print_fonface"
```

```
using Gaston
t = 0:0.01:1
set(font="CMU-Serif")
plot(t, sin.(2π*5*t), title = "A sine wave")
```
```
got unsupported keyword argument "font"
```

```
using Gaston
t = 0:0.01:1
plot(t, sin.(2π*5*t), title = "A sine wave",font="CMU-Serif")
```
```
got unsupported keyword argument "font"
```

Could you offer some demos? Thank you very much.

**Comment by mbaz** [comment]
This should work; notice the missing `t` in `font`:

`set(print_fontface="CMU-Serif")`

**Comment by ConAntares** [comment]
Thank your very much for your help.

But it also cannot change the font.

My code and result:
[cid:image002.png@01D506CC.543B5FD0]

Gnuplot version: 5.26
Gaston version: 0.9.2
Julia version: 1.10
OS: Windows10

Sincerely,
Luke

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by mbaz** [comment]
The image didn't make it through. Can you provide a link?

**Comment by ConAntares** [comment]
> 
> 
> The image didn't make it through. Can you provide a link?

I am sorry. The previous reply I sent in e-mail.

This is the picture:
![Codes and Result](https://github.com/ConAntares/Problems/blob/master/2019/20190510.png)

**Comment by mbaz** [comment]
I'm not sure what's going on, but that warning you got is not normal; a related bug is fixed in the master branch. Can you please try the same commands again in a fresh session? I just tried it myself here, and it works.

**Comment by ConAntares** [comment]
> 
> 
> I'm not sure what's going on, but that warning you got is not normal; a related bug is fixed in the master branch. Can you please try the same commands again in a fresh session? I just tried it myself here, and it works.

I have re-installed Gaston and tested it just now.
I got some errors as follows:
```
Testing Gaston
    Status `C:\Users\LukeNiu\AppData\Local\Temp\jl_A8F.tmp\Manifest.toml`
  [4b11ee91] Gaston v0.9.2
  [2a0f44e3] Base64  [`@stdlib/Base64`]
  [8bb1440f] DelimitedFiles  [`@stdlib/DelimitedFiles`]
  [8ba89e20] Distributed  [`@stdlib/Distributed`]
  [b77e0a4c] InteractiveUtils  [`@stdlib/InteractiveUtils`]
  [56ddb016] Logging  [`@stdlib/Logging`]
  [d6f4376e] Markdown  [`@stdlib/Markdown`]
  [a63ad114] Mmap  [`@stdlib/Mmap`]
  [9a3f8284] Random  [`@stdlib/Random`]
  [9e88b42a] Serialization  [`@stdlib/Serialization`]
  [6462fe0b] Sockets  [`@stdlib/Sockets`]
  [8dfed614] Test  [`@stdlib/Test`]
┌ Warning: Gnuplot returned an error message:
│
│ gnuplot> set output "C:\Users\LukeNiu\AppData\Local\Temp\/gastontest"
│                                                                       ^
│          line 0:          cannot open file; output not changed
│ system error: No such file or directory
│
│ GastonDone
│ )
└ @ Gaston C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\src\gaston_llplot.jl:180
┌ Warning: Gnuplot returned an error message:
│   GastonDone
│ )
└ @ Gaston C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\src\gaston_llplot.jl:180
Pass expected: Test Failed at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:162
  Expression: begin
        plot(1:10)
        printfigure()
        (Gaston.gnuplot_state).gp_error
    end == false
   Evaluated: true == false
Stacktrace:
 [1] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:162
 [2] top-level scope at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Test\src\Test.jl:1083
 [3] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:22
┌ Warning: Gnuplot returned an error message:
│
│ gnuplot> set output "C:\Users\LukeNiu\AppData\Local\Temp\/gastontest"
│                                                                       ^
│          line 0:          cannot open file; output not changed
│ system error: No such file or directory
│
│ GastonDone
│ )
└ @ Gaston C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\src\gaston_llplot.jl:180
┌ Warning: Gnuplot returned an error message:
│   GastonDone
│ )
└ @ Gaston C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\src\gaston_llplot.jl:180
┌ Warning: Gnuplot returned an error message:
│
│ gnuplot> set output "C:\Users\LukeNiu\AppData\Local\Temp\/gastontest"
│                                                                       ^
│          line 0:          cannot open file; output not changed
│ system error: No such file or directory
│
│ GastonDone
│ )
└ @ Gaston C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\src\gaston_llplot.jl:180
┌ Warning: Gnuplot returned an error message:
│   GastonDone
│ )
└ @ Gaston C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\src\gaston_llplot.jl:180
Pass expected: Test Failed at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:172
  Expression: begin
        plot(1:10)
        printfigure(term="eps")
        (Gaston.gnuplot_state).gp_error
    end == false
   Evaluated: true == false
Stacktrace:
 [1] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:172
 [2] top-level scope at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Test\src\Test.jl:1083
 [3] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:22
┌ Warning: Gnuplot returned an error message:
│
│ gnuplot> set output "C:\Users\LukeNiu\AppData\Local\Temp\/gastontest"
│                                                                       ^
│          line 0:          cannot open file; output not changed
│ system error: No such file or directory
│
│ GastonDone
│ )
└ @ Gaston C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\src\gaston_llplot.jl:180
┌ Warning: Gnuplot returned an error message:
│   GastonDone
│ )
└ @ Gaston C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\src\gaston_llplot.jl:180
Pass expected: Test Failed at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:177
  Expression: begin
        plot(1:10)
        printfigure(term="pdf") == 1
        (Gaston.gnuplot_state).gp_error
    end == false
   Evaluated: true == false
Stacktrace:
 [1] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:177
 [2] top-level scope at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Test\src\Test.jl:1083
 [3] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:22
┌ Warning: Gnuplot returned an error message:
│
│ gnuplot> set output "C:\Users\LukeNiu\AppData\Local\Temp\/gastontest"
│                                                                       ^
│          line 0:          cannot open file; output not changed
│ system error: No such file or directory
│
│ GastonDone
│ )
└ @ Gaston C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\src\gaston_llplot.jl:180
┌ Warning: Gnuplot returned an error message:
│   GastonDone
│ )
└ @ Gaston C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\src\gaston_llplot.jl:180
Pass expected: Test Failed at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:182
  Expression: begin
        plot(1:10)
        set(print_size="640,480")
        printfigure(term="svg")
        (Gaston.gnuplot_state).gp_error
    end == false
   Evaluated: true == false
Stacktrace:
 [1] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:182
 [2] top-level scope at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Test\src\Test.jl:1083
 [3] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:22
┌ Warning: Gnuplot returned an error message:
│   GastonDone
│ )
└ @ Gaston C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\src\gaston_llplot.jl:180
Test Summary: | Pass  Fail  Total
Pass expected |   57     4     61
ERROR: LoadError: Some tests did not pass: 57 passed, 4 failed, 0 errored, 0 broken.
in expression starting at C:\Users\LukeNiu\.julia\packages\Gaston\oUFi1\test\runtests.jl:21
ERROR: Package Gaston errored during testing
```

**Comment by mbaz** [comment]
I've never tested Gaston on Windows; I'll have to try it out. On Linux, all tests pass.

Have you tried to change the font in a fresh session?

**Comment by ConAntares** [comment]
> 
> 
> I've never tested Gaston on Windows; I'll have to try it out. On Linux, all tests pass.
> 
> Have you tried to change the font in a fresh session?

Yes, I have attempted many times.
I think the reason is caused by the incompatibility between the Windows10 and Gaston.

**Comment by mbaz** [comment]
It could be. Sorry about that. I develop and work in Linux. However, I'll take a look at this and try to fix any errors before the next release.

Maybe running Julia under WSL is a viable workaround?

**Comment by ConAntares** [comment]
> 
> 
> It could be. Sorry about that. I develop and work in Linux. However, I'll take a look at this and try to fix any errors before the next release.
> 
> Maybe running Julia under WSL is a viable workaround?

Thanks for your help very much! 

My school offered me a computer with Linux, I will try to use Gaston after the holidays.
I assert Gnuplot is more powerful than MatPloLlib, MATLAB, MATHEMATICA, PlotlyJS, and Tikz. I hope Gaston could become better and better~

Sincerely,
Lu

**Comment by mbaz** [comment]
You may want to follow this thread: https://discourse.julialang.org/t/process-streams-not-working-in-windows/24150/2

**Comment by asinghvi17** [comment]
It seems you're not actually on master - try `add Gaston#master` and then see if the solutions work?

When testing, you can see:
```julia
[4b11ee91] Gaston v0.9.2
```
whereas on master, you would see:
```julia
[4b11ee91] Gaston v0.9.2+ #master (https://github.com/mbaz/Gaston.jl.git)
```

That being said, I do get some test failures on OSX and `master` (though they appear to be unrelated).

```julia
   Testing Gaston
    Status `/var/folders/gl/tdt9v5vj4xqg5cdsz0ll1j7h0000gn/T/tmp6hF2rX/Manifest.toml`
  [4b11ee91] Gaston v0.9.2+ #master (https://github.com/mbaz/Gaston.jl.git)
  [2a0f44e3] Base64  [`@stdlib/Base64`]
  [8bb1440f] DelimitedFiles  [`@stdlib/DelimitedFiles`]
  [8ba89e20] Distributed  [`@stdlib/Distributed`]
  [b77e0a4c] InteractiveUtils  [`@stdlib/InteractiveUtils`]
  [56ddb016] Logging  [`@stdlib/Logging`]
  [d6f4376e] Markdown  [`@stdlib/Markdown`]
  [a63ad114] Mmap  [`@stdlib/Mmap`]
  [9a3f8284] Random  [`@stdlib/Random`]
  [9e88b42a] Serialization  [`@stdlib/Serialization`]
  [6462fe0b] Sockets  [`@stdlib/Sockets`]
  [8dfed614] Test  [`@stdlib/Test`]
Test Summary:           | Pass  Total
Figure and set commands |   25     25
2-D plots: Test Failed at /Users/anshul/.julia/packages/Gaston/n6aaf/test/runtests.jl:213
  Expression: begin
        set(reset=true)
        set(terminal="dumb", size="27,13")
        x = "\f                           \n  11 +-----------------+   \n  10 |-+ + + + + + + +*|   \n   9 |-+          ****-|   \n   7 |-+        **   +-|   \n   6 |-+      **     +-|   \n   5 |-+    **       +-|   \n   4 |-+  **         +-|   \n   3 |****           +-|   \n   1 |-+ + + + + + + +-|   \n   0 +-----------------+   \n     1 2 3 4 5 6 7 8 9 10  \n                           \n"
        a = repr("text/plain", plot(1:10))
        a == x
    end == true
   Evaluated: false == true
Stacktrace:
 [1] top-level scope at /Users/anshul/.julia/packages/Gaston/n6aaf/test/runtests.jl:213
 [2] top-level scope at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/Test/src/Test.jl:1083
 [3] top-level scope at /Users/anshul/.julia/packages/Gaston/n6aaf/test/runtests.jl:72
Test Summary: | Pass  Fail  Total
2-D plots     |   26     1     27
ERROR: LoadError: Some tests did not pass: 26 passed, 1 failed, 0 errored, 0 broken.
in expression starting at /Users/anshul/.julia/packages/Gaston/n6aaf/test/runtests.jl:71
ERROR: Package Gaston errored during testing

**Comment by ConAntares** [comment]
> 
> 
> It seems you're not actually on master - try `add Gaston#master` and then see if the solutions work?
> 
> When testing, you can see:
> 
> ```julia
> [4b11ee91] Gaston v0.9.2
> ```
> 
> whereas on master, you would see:
> 
> ```julia
> [4b11ee91] Gaston v0.9.2+ #master (https://github.com/mbaz/Gaston.jl.git)
> ```
> 
> That being said, I do get some test failures on OSX and `master` (though they appear to be unrelated).
> 
> ```julia
>    Testing Gaston
>     Status `/var/folders/gl/tdt9v5vj4xqg5cdsz0ll1j7h0000gn/T/tmp6hF2rX/Manifest.toml`
>   [4b11ee91] Gaston v0.9.2+ #master (https://github.com/mbaz/Gaston.jl.git)
>   [2a0f44e3] Base64  [`@stdlib/Base64`]
>   [8bb1440f] DelimitedFiles  [`@stdlib/DelimitedFiles`]
>   [8ba89e20] Distributed  [`@stdlib/Distributed`]
>   [b77e0a4c] InteractiveUtils  [`@stdlib/InteractiveUtils`]
>   [56ddb016] Logging  [`@stdlib/Logging`]
>   [d6f4376e] Markdown  [`@stdlib/Markdown`]
>   [a63ad114] Mmap  [`@stdlib/Mmap`]
>   [9a3f8284] Random  [`@stdlib/Random`]
>   [9e88b42a] Serialization  [`@stdlib/Serialization`]
>   [6462fe0b] Sockets  [`@stdlib/Sockets`]
>   [8dfed614] Test  [`@stdlib/Test`]
> Test Summary:           | Pass  Total
> Figure and set commands |   25     25
> 2-D plots: Test Failed at /Users/anshul/.julia/packages/Gaston/n6aaf/test/runtests.jl:213
>   Expression: begin
>         set(reset=true)
>         set(terminal="dumb", size="27,13")
>         x = "\f                           \n  11 +-----------------+   \n  10 |-+ + + + + + + +*|   \n   9 |-+          ****-|   \n   7 |-+        **   +-|   \n   6 |-+      **     +-|   \n   5 |-+    **       +-|   \n   4 |-+  **         +-|   \n   3 |****           +-|   \n   1 |-+ + + + + + + +-|   \n   0 +-----------------+   \n     1 2 3 4 5 6 7 8 9 10  \n                           \n"
>         a = repr("text/plain", plot(1:10))
>         a == x
>     end == true
>    Evaluated: false == true
> Stacktrace:
>  [1] top-level scope at /Users/anshul/.julia/packages/Gaston/n6aaf/test/runtests.jl:213
>  [2] top-level scope at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.1/Test/src/Test.jl:1083
>  [3] top-level scope at /Users/anshul/.julia/packages/Gaston/n6aaf/test/runtests.jl:72
> Test Summary: | Pass  Fail  Total
> 2-D plots     |   26     1     27
> ERROR: LoadError: Some tests did not pass: 26 passed, 1 failed, 0 errored, 0 broken.
> in expression starting at /Users/anshul/.julia/packages/Gaston/n6aaf/test/runtests.jl:71
> ERROR: Package Gaston errored during testing
> 
> Edit: cc @ConAntares
> ```

Thank you very much for your answer.
But in Windows, it also doesn't work and get errors when I test Gaston:

```Julia
Testing Gaston
    Status `C:\Users\LukeNiu\AppData\Local\Temp\jl_125E.tmp\Manifest.toml`
  [4b11ee91] Gaston v0.9.2+ #master (https://github.com/mbaz/Gaston.jl.git)
  [2a0f44e3] Base64  [`@stdlib/Base64`]
  [8bb1440f] DelimitedFiles  [`@stdlib/DelimitedFiles`]
  [8ba89e20] Distributed  [`@stdlib/Distributed`]
  [b77e0a4c] InteractiveUtils  [`@stdlib/InteractiveUtils`]
  [56ddb016] Logging  [`@stdlib/Logging`]
  [d6f4376e] Markdown  [`@stdlib/Markdown`]
  [a63ad114] Mmap  [`@stdlib/Mmap`]
  [9a3f8284] Random  [`@stdlib/Random`]
  [9e88b42a] Serialization  [`@stdlib/Serialization`]
  [6462fe0b] Sockets  [`@stdlib/Sockets`]
  [8dfed614] Test  [`@stdlib/Test`]
Test Summary:           | Pass  Total
Figure and set commands |   24     24
2-D plots: Error During Test at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\test\runtests.jl:213
  Test threw exception
  Expression: begin
        set(reset=true)
        set(terminal="dumb", size="27,13")
        x = "\f                           \n  11 +-----------------+   \n  10 |-+ + + + + + + +*|   \n   9 |-+          ****-|   \n   7 |-+        **   +-|
 \n   6 |-+      **     +-|   \n   5 |-+    **       +-|   \n   4 |-+  **         +-|   \n   3 |****           +-|   \n   1 |-+ + + + + + + +-|   \n   0 +-----------------+   \n     1 2 3 4 5 6 7 8 9 10  \n                           \n"
        a = repr("text/plain", plot(1:10))
        a == x
    end == true
  Gnuplot is taking too long to respond.
  Stacktrace:
   [1] error(::String) at .\error.jl:33
   [2] llplot(::Gaston.Figure) at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\src\gaston_llplot.jl:189
   [3] show(::Base.GenericIOBuffer{Array{UInt8,1}}, ::MIME{Symbol("text/plain")}, ::Gaston.Figure) at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\src\gaston_aux.jl:153
   [4] __binrepr at .\multimedia.jl:127 [inlined]
   [5] _binrepr at .\multimedia.jl:133 [inlined]
   [6] #repr#1(::Nothing, ::Function, ::MIME{Symbol("text/plain")}, ::Gaston.Figure) at .\multimedia.jl:115
   [7] (::getfield(Base, Symbol("#kw##repr")))(::NamedTuple{(:context,),Tuple{Nothing}}, ::typeof(repr), ::MIME{Symbol("text/plain")}, ::Gaston.Figure) at .\none:0
   [8] #repr#2(::Nothing, ::Function, ::String, ::Gaston.Figure) at .\multimedia.jl:116
   [9] repr(::String, ::Gaston.Figure) at .\multimedia.jl:116
   [10] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\test\runtests.jl:213
   [11] top-level scope at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Test\src\Test.jl:1083
   [12] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\test\runtests.jl:72
2-D plots: Error During Test at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\test\runtests.jl:220
  Test threw exception
  Expression: begin
        set(reset=true)
        set(terminal="ijulia")
        a = repr("text/plain", plot(1:10))
        a[1:35] == "<?xml version=\"1.0\" encoding=\"utf-8"
    end == true
  Gnuplot is taking too long to respond.
  Stacktrace:
   [1] error(::String) at .\error.jl:33
   [2] llplot(::Gaston.Figure) at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\src\gaston_llplot.jl:189
   [3] show(::Base.GenericIOBuffer{Array{UInt8,1}}, ::MIME{Symbol("text/plain")}, ::Gaston.Figure) at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\src\gaston_aux.jl:153
   [4] __binrepr at .\multimedia.jl:127 [inlined]
   [5] _binrepr at .\multimedia.jl:133 [inlined]
   [6] #repr#1(::Nothing, ::Function, ::MIME{Symbol("text/plain")}, ::Gaston.Figure) at .\multimedia.jl:115
   [7] (::getfield(Base, Symbol("#kw##repr")))(::NamedTuple{(:context,),Tuple{Nothing}}, ::typeof(repr), ::MIME{Symbol("text/plain")}, ::Gaston.Figure) at .\none:0
   [8] #repr#2(::Nothing, ::Function, ::String, ::Gaston.Figure) at .\multimedia.jl:116
   [9] repr(::String, ::Gaston.Figure) at .\multimedia.jl:116
   [10] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\test\runtests.jl:220
   [11] top-level scope at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Test\src\Test.jl:1083
   [12] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\test\runtests.jl:72
2-D plots: Error During Test at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\test\runtests.jl:230
  Test threw exception
  Expression: begin
        ac = Gaston.AxesConf(title="T")
        (x1, exp_pdf) = Gaston.hist(randn(10000), 25)
        exp_pdf .= exp_pdf ./ (step(x1) * sum(exp_pdf))
        exp_cconf = Gaston.CurveConf(plotstyle="boxes", linecolor="blue", legend="E")
        exp_curve = Gaston.Curve(x1, exp_pdf, exp_cconf)
        x2 = -5:0.05:5
        theo_pdf = #= C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\test\runtests.jl:239 =# @__dot__((1 / sqrt(2π)) * exp(-(x2 ^ 2) / 2))
        theo_cconf = Gaston.CurveConf(linecolor="black", legend="T")
        theo_curve = Gaston.Curve(x2, theo_pdf, theo_cconf)
        figure(1)
        Gaston.push_figure!(1, ac, exp_curve, theo_curve)
        figure(1)
    end == 1
  Gnuplot is taking too long to respond.
  Stacktrace:
   [1] error(::String) at .\error.jl:33
   [2] llplot(::Gaston.Figure) at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\src\gaston_llplot.jl:189
   [3] show(::Base.TTY, ::MIME{Symbol("text/plain")}, ::Gaston.Figure) at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\src\gaston_aux.jl:153
   [4] display(::TextDisplay, ::MIME{Symbol("text/plain")}, ::Any) at .\multimedia.jl:204
   [5] display(::TextDisplay, ::Any) at .\multimedia.jl:205
   [6] display(::Any) at .\multimedia.jl:287
   [7] #figure#14(::Bool, ::Function, ::Int64) at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\src\gaston_hilvl.jl:75
   [8] figure(::Int64) at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\src\gaston_hilvl.jl:50
   [9] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\test\runtests.jl:230
   [10] top-level scope at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.1\Test\src\Test.jl:1083
   [11] top-level scope at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\test\runtests.jl:72
Test Summary: | Pass  Error  Total
2-D plots     |   24      3     27
ERROR: LoadError: Some tests did not pass: 24 passed, 0 failed, 3 errored, 0 broken.
in expression starting at C:\Users\LukeNiu\.julia\packages\Gaston\n6aaf\test\runtests.jl:71
ERROR: Package Gaston errored during testing
```

**Comment by mbaz** [comment]
@ConAntares Indeed, Gaston master does not work with windows, but the cause is different than for v0.9.2.

@asinghvi17 Thanks for the help and the report. That particular test seems to be quite problematic; I've found that different gnuplot versions generate slightly different plots. Gnuplot's configuration can also affect it. I will remove the test for the time being. If that is the only failure you get, you should be OK.

**Comment by ConAntares** [comment]
> 
> 
> @ConAntares Indeed, Gaston master does not work with windows, but the cause is different than for v0.9.2.
> 
> @asinghvi17 Thanks for the help and the report. That particular test seems to be quite problematic; I've found that different gnuplot versions generate slightly different plots. Gnuplot's configuration can also affect it. I will remove the test for the time being. If that is the only failure you get, you should be OK.

There are a lot of users using Windows. Will you support for windows in future version?

**Comment by mbaz** [comment]
I want to support windows. My problem right now (see the thread I linked above) is that the current issue with streams will take some time to debug. I don't know if the problem lies with gnuplot or with Julia. I will do my best, but it make take some time. I want to release v0.9.3 soon, so I might have to temporarily disable support for windows. I'm sorry for the bad situation at the moment.

**Comment by mbaz** [comment]
Just a quick note -- I'm making progress on this front. I'm confident I'll be able to support Windows soon.

**Comment by mbaz** [comment]
Current Gaston master passes all tests on Windows, and the font commands as described above seem to work.

Note that plotting in Windows can be slow; see https://discourse.julialang.org/t/standard-streams-much-slower-in-windows-than-in-linux/24924

**Comment by ConAntares** [comment]
> 
> 
> Current Gaston master passes all tests on Windows, and the font commands as described above seem to work.
> 
> Note that plotting in Windows can be slow; see https://discourse.julialang.org/t/standard-streams-much-slower-in-windows-than-in-linux/24924
> 
> I'll wait to close this bug until I make a release; all I need to do is finish the new documentation.

I'm very happy about this. Thank you for your support.

**Comment by mbaz** [comment]
Thanks! v0.10.0 is around the corner, I'm almost done with the docs.

**Comment by mbaz** [comment]
Fixed in v0.10.0. See new docs: https://mbaz.github.io/Gaston.jl/v0.10.0/

**Comment by ConAntares** [comment]
> 
> 
> Fixed in v0.10.0. See new docs: https://mbaz.github.io/Gaston.jl/v0.10.0/

It works well on my computer. I will use Gaston in my articles. And I will recommend it to others.
